book/src/languages.md

```diff
@@ -14,6 +14,18 @@ name = "rust"
 auto-format = false
 ```
 
+## LSP formatting options
+
+Use `format` field to pass extra formatting options to [Document Formatting Requests](https://github.com/microsoft/language-server-protocol/blob/gh-pages/_specifications/specification-3-16.md#document-formatting-request--leftwards_arrow_with_hook).
+
+```toml
+[[language]]
+name = "typescript"
+auto-format = true
+# pass format options according to https://github.com/typescript-language-server/typescript-language-server#workspacedidchangeconfiguration omitting the "[language].format." prefix.
+format = { "semicolons" = "insert", "insertSpaceBeforeFunctionParenthesis" = true }
```

Comment:
This was a mistake we did with language server configuration: I'd prefer `format` was nested under the `language-server` key. Though maybe we can keep it toplevel

Reply:
It would make sense inside the language server configuration, especially if we ever plan to support multiple LS per language. I have seen neovim setups where multiple LS get attached to a buffer, mostly for chaining formatting functionality. Can't say that i liked it. Have you guys thought in that direction? I haven't noticed the topic here, but i am just getting into the project.

Reply:
https://github.com/helix-editor/helix/issues/1396

Reply:
i guess we need this. 6c29b0e79cc98ec73bce67154826127253546533 nests it inside the lsp config now.